### ISSUE TITLE
Regression: systemd-gpt-auto-generator causes boot delays and intermittent boot failure.

### ISSUE THREAD

**Issue by asavah** [systemd/systemd#11205]
**systemd version the issue has been seen with**
master @ https://github.com/systemd/systemd/commit/a361cc99ae9f67ed0ec99f3895abd16796dd3319 with #10268 reverted (see explanation below)

**Used distribution**
own pet OS, x86_64, linux 4.19.10, UEFI GPT on nvme

**Expected behaviour you didn't see**
Clean and fast boot.

**Unexpected behaviour you saw**
Boot delay about 180 seconds, intermittent boot failures (emergency shell)

**Steps to reproduce the problem**
Boot the OS on an UEFI system with GPT layout.

This one is a little tricky because currently there are 2 issues which may impede proper booting.
The first one is being tackled at #11178 , to avoid possible interference from the first issue I reverted #10268 to make sure that this is entirely new issue.

The symptoms are as follows system starts booting and after the "banner" it hangs for 90secs , even keyboard numlock led won't respond. After that system continues booting and stops for another 90 secs, after that the system may either boot normally or fail (emergency shell).

I can provide a link to the log however the interesting things can't be seen in the log ...

(sorry for the screenshot made with the phone, the mobo lacks a proper UART port so capturing boot log is impossible ...)

On the pic the moment when system "hangs" is captured:

![photo_2018-12-18_21-46-19](https://user-images.githubusercontent.com/5447220/50178930-704c8800-030e-11e9-9b15-e055c653ea76.jpg)

```
root@msib450ig:/usr/lib/systemd/system-generators# ls -alh
total 240K
drwxr-xr-x  2 root root 4.0K Dec 18 21:49 .
drwxr-xr-x 15 root root 4.0K Dec 18 19:54 ..
lrwxrwxrwx  1 root root   22 Dec 16 21:55 netplan -> ../../netplan/generate
-rwxr-xr-x  1 root root  31K Dec 16 21:55 nfs-server-generator
-rwxr-xr-x  1 root root  19K Dec 16 21:55 rpc-pipefs-generator
-rwxr-xr-x  1 root root 9.9K Dec 18 20:35 systemd-bless-boot-generator
-rwxr-xr-x  1 root root  27K Dec 18 20:35 systemd-cryptsetup-generator
-rwxr-xr-x  1 root root  15K Dec 18 20:35 systemd-debug-generator
-rwxr-xr-x  1 root root  35K Dec 18 20:35 systemd-fstab-generator
-rwxr-xr-x  1 root root  14K Dec 18 20:35 systemd-getty-generator
-rwxr-xr-x  1 root root  26K Dec 18 20:35 systemd-gpt-auto-generator
-rwxr-xr-x  1 root root  15K Dec 18 20:35 systemd-run-generator
-rwxr-xr-x  1 root root 9.9K Dec 18 20:35 systemd-system-update-generator
-rwxr-xr-x  1 root root  15K Dec 18 20:35 systemd-veritysetup-generator
```

From the pic and the directory listing it can be seen that it's waiting on systemd-gpt-auto-generator or nfs-server-generator (this machine does not act as NFS server).

**Deleting systemd-gpt-auto-generator works around the issue!**

Edit: added current fstab for reference
```
root@msib450ig:/home# cat /etc/fstab
# fstab automatically generated for msib450ig
#<dev>  <mount point>   <type>  <options>       <dump>  <pass>
/dev/nvme0n1p1  /boot/efi       vfat    umask=0077      0       0
/dev/nvme0n1p2  /       ext4    defaults,noatime,discard        0       0
/dev/nvme0n1p3  /home   ext4    defaults,noatime,discard        0       0
```
```
root@msib450ig:/home# fdisk -l
Disk /dev/nvme0n1: 223.6 GiB, 240057409536 bytes, 468862128 sectors
Disk model: Force MP510
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: 4D2F2655-2E94-FC4E-946B-82B98035BF5A

Device            Start       End   Sectors   Size Type
/dev/nvme0n1p1     2048   1026047   1024000   500M EFI System
/dev/nvme0n1p2  1026048  84912127  83886080    40G Linux filesystem
/dev/nvme0n1p3 84912128 468862094 383949967 183.1G Linux filesystem
```

I won't have time to properly bisect this, but I'm certain that @ https://github.com/systemd/systemd/commit/bd1c8360a47451341d261b877629693e37233c4c this is issue was not present.

**Comment by poettering** [comment]
Ah, right, the dissector waits for udev now, but while the generator runs udev isn't around... we should assume that everything is probed in that case.

**Comment by poettering** [comment]
Fix waiting in #11215

**Comment by keszybz** [comment]
So it's a regression from a8040b6d0a. A flag for `dissect_image()` like DISSECT_IMAGE_NO_UDEV ? It'd mean that we wait for the kernel, but not udev, and cannot actually make use of the devices.
